test/vanilla/legacy/Expected/AcceptanceTests/MediaTypes/mediatypes/operations/_media_types_client_operations.py

```diff
@@ -46,10 +46,11 @@ def build_analyze_body_request(
     url = kwargs.pop("template_url", '/mediatypes/analyze')
 
     # Construct headers
-    header_parameters = kwargs.pop("headers", {})  # type: Dict[str, Any]
+    header_parameters = {}  # type: Dict[str, Any]
     if content_type is not None:
         header_parameters['Content-Type'] = _SERIALIZER.header("content_type", content_type, 'str')
     header_parameters['Accept'] = _SERIALIZER.header("accept", accept, 'str')
```

Comment:
There is a subtle problem with this approach -- it only works if the user provides the "Accept" header using the same casing as used here.  HTTP headers are by definition case-insensitive [ref: [RFC 2616, Section 4.2](https://datatracker.ietf.org/doc/html/rfc2616#section-4.2)]. So if the user passed "accept" in the headers dict, the request would wind up with an "accept" header _and_ and "Accept" header.

So I think you need to check the user headers for a _case-insensitive_ match on the key before setting a header parameter explicitly, something like:
```suggestion
    if len([x for x in headers.keys() if x.lower() == "accept"]) == 0:
        header_parameters['Accept'] = _SERIALIZER.header("accept", accept, 'str')
```